src/overlays/actors/ovl_En_Mu/z_en_mu.c

```diff
@@ -44,27 +34,179 @@ static ColliderCylinderInit D_80AB0BD0 = {
     },
     { 100, 70, 0, { 0, 0, 0 } },
 };
-*/
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Mu/func_80AB0420.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Mu/func_80AB0428.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Mu/func_80AB0584.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Mu/func_80AB05C4.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Mu/EnMu_Init.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Mu/EnMu_Destroy.s")
+CollisionCheckInfoInit2 D_80AB0BFC = { 0x00, 0x0000, 0x0000, 0x0000, 0xFF };
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Mu/func_80AB0724.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Mu/EnMu_Update.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Mu/func_80AB08A4.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Mu/func_80AB0994.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Mu/func_80AB09A8.s")
+const ActorInit En_Mu_InitVars = {
+    ACTOR_EN_MU,
+    ACTORCAT_NPC,
+    FLAGS,
+    OBJECT_MU,
+    sizeof(EnMu),
+    (ActorFunc)EnMu_Init,
+    (ActorFunc)EnMu_Destroy,
+    (ActorFunc)EnMu_Update,
+    (ActorFunc)EnMu_Draw,
+};
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Mu/EnMu_Draw.s")
+void EnMu_SetActionFunction(Actor* thisx, GlobalContext* globalCtx) {
+    EnMu* this = THIS;
+    this->actionFunc = globalCtx;
+}
+
+void EnMu_Interact(EnMu* this, GlobalContext* globalCtx) {
+    u8 textByteOffset[] = { 0x42, 0x43, 0x3F, 0x41, 0x3E };
+    u8 twoPower[] = { 0x01, 0x02, 0x04, 0x08, 0x10 };
+    u8 textFlags;
+    s32 randomIndex;
+    s32 i;
+
+    textFlags = gSaveContext.eventInf[2] & 0x1F;
+    gSaveContext.eventInf[2] &= 0xFFE0;
+    randomIndex = (globalCtx->state.frames + (s32)(Rand_ZeroOne() * 5.0f)) % 5;
+
+    for (i = 0; i < 5; i++) {
+
+        if ((textFlags & twoPower[randomIndex]) == 0) {
+            break;
+        }
+
+        if (++randomIndex >= 5) {
+            randomIndex = 0;
+        }
+    }
+
+    if (i == 5) {
+        if (this->defFaceReaction == (textByteOffset[randomIndex] | 0x7000)) {
+            if (++randomIndex >= 5) {
+                randomIndex = 0;
+            }
+        }
+        textFlags = 0;
+    }
+
+    textFlags |= twoPower[randomIndex];
+    this->defFaceReaction = textByteOffset[randomIndex] | 0x7000;
+    textFlags &= 0xFF;
+    gSaveContext.eventInf[2] |= textFlags;
+}
+
+u16 EnMu_GetFaceReaction(GlobalContext* globalCtx, EnMu* this) {
```

Comment:
```suggestion
u16 EnMu_GetFaceReaction(GlobalContext* globalCtx, Actor* thisx) {
```
I think this should take thisx for the func_800343CC callback